include/meson.build

```diff
@@ -0,0 +1,23 @@
+header_files = [
+    'vir/detail.h',
+    'vir/simd_bit.h',
+    'vir/simd_bitset.h',
+    'vir/simd_cast.h',
+    'vir/simd_float_ops.h',
+    'vir/simd_resize.h',
+    'vir/simd.h',
+    'buffer_skeleton.hpp',
+    'buffer.hpp',
+    'circular_buffer.hpp',
+    'claim_strategy.hpp',
+    'graph.hpp',
+    'node_traits.hpp',
+    'node.hpp',
+    'port_traits.hpp',
+    'port.hpp',
+    'sequence.hpp',
+    'typelist.hpp',
+    'utils.hpp',
+    'wait_strategy.hpp'
+]
```

Comment:
This just defines where the headers have to be installed, but not their dependencies, which have to be specified along with the include path everywhere it is used (in this project that's just the tests and benchmarks, but looking at gnuradio and the modularisation we talked about, there will be more dependencies between subprojects).
In CMake this is solved for our header-only modules by defining INTERFACE targets, which bundle the include directories and their dependencies into a single target.
It seems that the meson way of doing this would be to define an internal dependency with [`declare_dependency()`](https://mesonbuild.com/Reference-manual_functions.html#declare_dependency) here.

Something like:

``` meson
graph_dep = declare_dependencies(include_directories : ".", dependencies : [fmt_dep, reflcpp_dep])
```

then in the tests/benchmarks we just specify `deps : [graph_dep, ut_dep]`. This also allows to restructure the project into sub-dependencies (e.g. use an external version of vir_simd) or move parts around without things falling apart or needing changes everywhere.